src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -37,10 +38,22 @@ namespace winrt::Microsoft::Management::Deployment::implementation
     {
         co_return Connect();
     }
+    winrt::Microsoft::Management::Deployment::ConnectResult GetConnectCatalogErrorResult()
+    {
+        auto connectResult = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::ConnectResult>>();
+        connectResult->Initialize(winrt::Microsoft::Management::Deployment::ConnectResultStatus::CatalogError, nullptr);
+        return *connectResult;
+    }
     winrt::Microsoft::Management::Deployment::ConnectResult PackageCatalogReference::Connect()
     {
         try
         {
+            if (FAILED(EnsureComCallerHasCapability(Capability::PackageQuery)))
+            {
+                // TODO: When more error codes are added, this should go back as something other than CatalogError.

```

Comment:
Why not `E_ACCESSDENIED` right now?

Reply:
To throw? Would prefer not to throw. Needs an idl update to return more errors unfortunately.